security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/ActionDeserializer.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.json;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.powsybl.commons.json.JsonUtil;
+import com.powsybl.security.action.Action;
+import com.powsybl.security.action.MultipleActionsAction;
+import com.powsybl.security.action.SwitchAction;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ */
+public class ActionDeserializer extends StdDeserializer<Action> {
+
+    public ActionDeserializer() {
+        super(Action.class);
+    }
+
+    private static class ParsingContext {
+        String type;
+        String id;
+        String switchId;
+        Boolean open;
+        List<Action> actions;
+    }
+
+    @Override
+    public Action deserialize(JsonParser parser, DeserializationContext deserializationContext) throws IOException {
+        ParsingContext context = new ParsingContext();
+        JsonUtil.parseObject(parser, fieldName -> {
+            switch (fieldName) {
+                case "type":
+                    context.type = parser.nextTextValue();
+                    return true;
+                case "id":
+                    context.id =  parser.nextTextValue();
+                    return true;
+                case "switchId":
+                    context.switchId =  parser.nextTextValue();
+                    return true;
+                case "open":
+                    parser.nextToken();
+                    context.open =  parser.getBooleanValue();
+                    return true;
+                case "actions":
+                    parser.nextToken();
+                    context.actions = parser.readValueAs(new TypeReference<ArrayList<Action>>() {
+                    });
+                    return true;
+                default:
+                    return false;
+            }
+        });
+        switch (Objects.requireNonNull(context.type)) {
```

Comment:
We should raise a more expressive exception than an NPE if the type is absent:
probably a jackson exception which provides information about the place of the problem in the JSON